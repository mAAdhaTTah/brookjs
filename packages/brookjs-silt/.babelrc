{
  "env": {
    "test": {
      "presets": [
        ["brookjs", { "loose": false, "modules": "commonjs", "pragma": "h" }]
      ]
    },
    "commonjs": {
      "presets": [
        ["brookjs", { "loose": false, "modules": "commonjs", "pragma": "h" }]
      ]
    },
    "es": {
      "presets": [
        ["brookjs", { "loose": false, "modules": false, "pragma": "h" }]
      ]
    }
  }
}
