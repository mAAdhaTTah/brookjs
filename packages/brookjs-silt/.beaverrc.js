export const dir = 'src';

/**
 * Mocha testing configuration.
 */
export const mocha = {
  reporter: 'spec',
  ui: 'bdd',
  requires: [
    '../../setupTests.ts',
    '@testing-library/react',
    'jsdom-global/register',
    'raf/polyfill'
  ]
};

/**
 * Webpack build configuration.
 */
export const webpack = {
  entry: {
    app: 'app.js'
  },
  output: {
    path: 'dist/',
    filename: '[name].js'
  },
  modifier: x => x
};
