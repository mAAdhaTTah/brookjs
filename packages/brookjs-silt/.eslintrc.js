const path = require('path');

module.exports = {
  root: true,
  parser: 'typescript-eslint-parser',
  extends: '../../.eslintrc.js',
  plugins: ['typescript'],
  rules: {
    indent: [2, 2],
    'no-unused-vars': 0,

    // TypeScript
    'typescript/no-unused-vars': [2]
  },
  settings: {
    'import/resolver': {
      node: {
        extensions: ['.js', '.ts', '.tsx']
      }
    }
  }
};
