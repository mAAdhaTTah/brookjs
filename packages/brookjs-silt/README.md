 # brookjs-silt
 
A React extension for building streaming web components.
