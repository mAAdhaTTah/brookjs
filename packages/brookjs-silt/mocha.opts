--reporter spec
--ui bdd
--require babel-register
src/**/__tests__/*.spec.js
