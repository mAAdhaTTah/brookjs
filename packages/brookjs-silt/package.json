{
  "name": "brookjs-silt",
  "version": "0.11.2",
  "description": "Handlebars-based rendering engine, built on diffhtml.",
  "main": "cjs/index.js",
  "private": true,
  "scripts": {
    "test": "npm run test:lint && npm run test:type && npm run test:unit",
    "test:lint": "eslint src --cache true --fix",
    "test:type": "flow check",
    "test:unit": "beaver test unit --coverage true --env test",
    "test:int": "cross-env NODE_ENV=test mocha --require babel-register int/run.js",
    "dev": "beaver dev tdd --env test",
    "clean": "rimraf cjs dist es coverage",
    "build": "npm run build:commonjs && npm run build:es && npm run build:umd && npm run build:umd:min",
    "build:commonjs": "cross-env NODE_ENV=commonjs babel src --ignore __tests__ --out-dir cjs",
    "build:es": "cross-env NODE_ENV=es babel src --ignore __tests__ --out-dir es",
    "build:umd": "cross-env NODE_ENV=es webpack src/index.js dist/brook.js",
    "build:umd:min": "cross-env NODE_ENV=es webpack src/index.js dist/brook.min.js",
    "prepublishOnly": "npm run clean && npm test && npm run build"
  },
  "author": "James DiGioia <jamesorodig@gmail.com> (http://jamesdigioia.com)",
  "license": "MIT",
  "devDependencies": {
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.0",
    "babel-eslint": "^8.0.0",
    "babel-loader": "^7.1.2",
    "babel-plugin-syntax-flow": "^6.18.0",
    "babel-plugin-transform-flow-strip-types": "^6.22.0",
    "babel-plugin-transform-react-jsx": "^6.24.1",
    "babel-preset-brookjs": "^0.11.1",
    "babel-register": "^6.26.0",
    "brookjs-cli": "^0.11.2",
    "chai": "^4.1.2",
    "cross-env": "^5.0.5",
    "eslint": "^4.7.0",
    "eslint-config-valtech": "^1.1.0",
    "eslint-import-resolver-node": "^0.3.1",
    "eslint-plugin-flowtype": "^2.35.1",
    "eslint-plugin-import": "^2.7.0",
    "eslint-plugin-react": "^7.5.1",
    "flow-bin": "^0.61.0",
    "react-dom": "^16.2.0",
    "rimraf": "^2.6.2",
    "webpack": "^3.6.0",
    "webpack-merge": "^4.1.0"
  },
  "dependencies": {
    "brookjs": "^0.11.2",
    "ramda": "^0.25.0",
    "react": "^16.2.0"
  }
}
