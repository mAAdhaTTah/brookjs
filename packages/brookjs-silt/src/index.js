// @flow
import Aggregator from './Aggregator';
import Collector from './Collector';
import fromReact from './fromReact';
import h from './h';
import loop from './loop';
import observableOf from './observableOf';
import view from './view';
import withRef$ from './withRef$';

export { Aggregator, Collector, fromReact, h,
    h as createElement, loop, observableOf,
    view, withRef$ };
