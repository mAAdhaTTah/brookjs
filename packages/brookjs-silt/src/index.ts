import RootJunction from './RootJunction';

export * from './toJunction';
export * from './useDeltas';
export * from './withRef';
export { RootJunction };
