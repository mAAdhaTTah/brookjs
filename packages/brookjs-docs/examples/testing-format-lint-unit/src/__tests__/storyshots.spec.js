import initStoryshots from '@storybook/addon-storyshots';
import { render } from '@testing-library/react';
import { RootJunction } from 'brookjs-silt';

const reactTestingLibrarySerializer = {
  print: (val, serialize) => serialize(val.container.firstChild),
  test: val => val && val.hasOwnProperty('container')
};

initStoryshots({
  framework: 'react',
  renderer: element =>
    render(element, {
      wrapper: RootJunction
    }),
  snapshotSerializers: [reactTestingLibrarySerializer]
});
