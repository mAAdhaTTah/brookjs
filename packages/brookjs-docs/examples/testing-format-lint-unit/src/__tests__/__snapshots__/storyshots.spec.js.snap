// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots App Empty List 1`] = `
<main>
  <div
    class="add-todo"
  >
    <label
      class="input"
      for="input"
    >
      Todo: 
      <input
        id="input"
        name="input"
        value=""
      />
    </label>
    <button
      class="btn"
    >
      Add Todo
    </button>
  </div>
  <ul />
</main>
`;

exports[`Storyshots App With Todos 1`] = `
<main>
  <div
    class="add-todo"
  >
    <label
      class="input"
      for="input"
    >
      Todo: 
      <input
        id="input"
        name="input"
        value=""
      />
    </label>
    <button
      class="btn"
    >
      Add Todo
    </button>
  </div>
  <ul>
    <li>
      First
    </li>
    <li>
      Second
    </li>
    <li>
      Third
    </li>
  </ul>
</main>
`;

exports[`Storyshots Button Basic 1`] = `
<button
  class="btn"
>
  Click me
</button>
`;

exports[`Storyshots Input Basic 1`] = `
<label
  class="input"
  for="input"
>
  Input
  <input
    id="input"
    name="input"
    value=""
  />
</label>
`;
